include/deal.II/base/parameter_handler.h

```diff
@@ -125,6 +125,90 @@ namespace Patterns
      * assumption significantly, you can still overload this function.
      */
     virtual std::size_t memory_consumption () const;
+
+
+    /**
+     * Convert a compatible value type to a string. If the type is not
+     * compatible, an excepion ExcIncompatibleType is thrown.
```

Comment:
That's too restrictive. `Patterns::List` now supports `std::vector<T>`, with `T` = `int`, `unsigned int`, `double`, `Point<1>`, `Point<2>`, and many others. Which type do you want to typedef in Patterns::List? I do this by checking `typeid(T)` for all compatible types...